pkg/server/serveroption/option.go

```diff
@@ -26,14 +34,45 @@ var Default = Options{
 type Options struct {
 	// Both sizes should really be uint32 but it's better saved for a single
 	// refactor commit.
-
 	MaxFlows      int // max number of flows that can be stored in the ring buffer
 	MonitorBuffer int // buffer size for monitor payload
+
+	CiliumDaemon interface{} // when running inside Cilium, contains a reference to the daemon
```

Comment:
Does it have to be an empty interface?

Reply:
The underlying type here will be `github.com/cilium/cilium/daemon/cmd.Daemon`. I opted for an interface instead of the concrete type for two reasons:

- Referencing the concrete type would basically pull in all packages of cilium-agent into `vendor/` (since `Daemon` is the root)
- The concrete as it is defined today is impossible to mock

Unfortunately, at least to me, there is no obvious minimal interface that `Daemon` would satisfy. This type has many ad-hoc methods and getters that make it unpractical to implement a coherent interface. Each hook will want to access a different aspect of Daemon and therefore downcast to a custom interface.

This is what all consumers of `Daemon` inside Cilium already do. To give you three random examples of the various interfaces that `Daemon` implements (there are many more):

- https://github.com/cilium/cilium/blob/cfb7333a064e4bf00ee4bcf95e556b5331b632cd/pkg/datapath/ipcache/listener.go#L49-L51
- https://github.com/cilium/cilium/blob/f0d65d667e33f8a137606015cc37c8d4d3d4e95c/pkg/identity/cache/allocator.go#L106-L116
- https://github.com/cilium/cilium/blob/ce2079869a0385423e802ca22c9233bda31dd93a/pkg/k8s/watchers/watcher.go#L113-L117

Additional examples include the five `pkg/parser/getter` interfaces in Hubble, which are also all references to `Daemon` at runtime.

Reply:
Ideally it would be something more than just empty interface. Even it's an interface with one or two methods that is deinfed in *this* pacakge. 

Perhaps something like version? Basic config? Mode? Etc...

Reply:
Please feel free to propose a concrete method in that struct. I didn't find any.

Reply:
Maybe start with something as basic as this? https://github.com/cilium/cilium/blob/master/daemon/cmd/daemon.go#L149

Reply:
Sounds good! I will add it.